test/controllers/sdes/SdesCallbackControllerSpec.scala

```diff
@@ -0,0 +1,124 @@
+
+  val mockFileDetailsRepository: FileDetailsRepository = mock[FileDetailsRepository]
+  val mockEmailService: EmailService                   = mock[EmailService]
+
+  override def beforeEach(): Unit = {
+    reset(mockFileDetailsRepository, mockEmailService)
+    super.beforeEach()
+  }
+
+  val application: Application = applicationBuilder()
+    .overrides(
+      bind[FileDetailsRepository].toInstance(mockFileDetailsRepository),
+      bind[EmailService].toInstance(mockEmailService)
+    )
+    .build()
+
+  "SdesCallbackController" - {
+    "must return Ok for success file notification" in {
+
+      forAll(arbitrarySuccessSdesCallback.arbitrary, arbitrary[FileDetails]) { (sdesCallback, fileDetails) =>
+        when(mockFileDetailsRepository.findByConversationId(sdesCallback.correlationID)).thenReturn(Future.successful(Some(fileDetails)))
+
+        val request = FakeRequest(POST, routes.SdesCallbackController.callback.url).withBody(Json.toJson(sdesCallback))
+        val result  = route(application, request).value
+        status(result) mustEqual OK
+
+      }
+    }
+
+    "must return Ok for failure notification and update status and send email" in {
+      forAll(arbitraryFailureSdesCallback.arbitrary, arbitraryPendingFileDetails.arbitrary) { (sdesCallback, fileDetails) =>
+        val updatedStatus = sdesCallback.failureReason match {
+          case Some(reason) if reason.toLowerCase.contains("virus") => RejectedSDESVirus
+          case _                                                    => RejectedSDES
+        }
+        when(mockFileDetailsRepository.findByConversationId(sdesCallback.correlationID)).thenReturn(Future.successful(Some(fileDetails)))
+        when(mockFileDetailsRepository.updateStatus(sdesCallback.correlationID.value, updatedStatus)).thenReturn(Future.successful(Some(fileDetails)))
+        when(
+          mockEmailService.sendAndLogEmail(any[String], any[String], any[String], any[Option[AgentContactDetails]], any[Boolean], any[ReportType])(
+            any[HeaderCarrier]
+          )
+        )
+          .thenReturn(Future.successful(Seq(ACCEPTED)))
+        val request = FakeRequest(POST, routes.SdesCallbackController.callback.url).withBody(Json.toJson(sdesCallback))
+        val result  = route(application, request).value
+
+        status(result) mustEqual OK
+        verify(mockFileDetailsRepository).updateStatus(sdesCallback.correlationID.value, updatedStatus)
+        verify(mockEmailService, atLeast(1)).sendAndLogEmail(any[String],
+                                                             any[String],
+                                                             any[String],
+                                                             any[Option[AgentContactDetails]],
+                                                             any[Boolean],
+                                                             any[ReportType]
```

Comment:
Do we need to check that the arguments passed to `sendAndLogEmail` are as expected?